src/haddock/modules/topology/topocg/aa2cg.py

```diff
@@ -399,7 +400,27 @@ def extract_groups(pair_list):
     out.close()
 
 
-def determine_ss(structure, skipss, pdbf_path):
+def create_file_with_cryst(pdb_file, pdb_file_copy):
+    """
+    This function creates a new pdb because the CRYST line is missing from the pdf file.     
+    This line is necessary for DSSP.
+
+    Args:
+        pdb_file: str
+        pdb_file_copy: str
+
+    Returns:
+
+    """
+    file_in = open(pdb_file, "r")
+    file_out = open(pdb_file_copy, "w")
+    content = file_in.read()
+    file_out.write(CRYST_LINE + content)
+    file_in.close()
+    file_out.close()
```

Comment:
It's best to use the `with` construction when handling files - it automatically ensures the files are closed even if there is any error, its overall better resource management.

Here since you have two opens, you can use both in one line;


```python
with open(pdb_file, "r") as file_in, open(pdb_file_copy, "w") as file_out:
    content = file_in.read()
    file_out.write(CRYST_LINE + content)
```

Reply:
yes, I did this function quickly, I will correct it